crates/uv-installer/src/compile.rs

```diff
@@ -85,12 +86,34 @@ pub async fn compile_tree(
     debug!("Starting {} bytecode compilation workers", worker_count);
     let mut worker_handles = Vec::new();
     for _ in 0..worker_count.get() {
-        worker_handles.push(tokio::task::spawn(worker(
+        let (tx, rx) = oneshot::channel();
+
+        let worker = worker(
             dir.to_path_buf(),
             python_executable.to_path_buf(),
             pip_compileall_py.clone(),
             receiver.clone(),
-        )));
+        );
+
+        // Spawn each worker on a dedicated thread.
+        std::thread::Builder::new()
+            .name("uv-compile".to_owned())
+            .spawn(move || {
+                // Report panics back to the main thread.
+                let result = panic::catch_unwind(AssertUnwindSafe(|| {
+                    tokio::runtime::Builder::new_current_thread()
+                        .enable_all()
+                        .build()
+                        .expect("Failed to build runtime")
+                        .block_on(worker)
```

Comment:
Does it still makes sense for the worker to be async? We're not really using any async-specific features there except for timeouts.

Reply:
I looked into making them non-async very briefly, but getting the timeouts to work with synchronous I/O is not trivial so I left it for now.